specs/file_hierarchy_for_the_verification_of_os_artifacts.md

```diff
@@ -0,0 +1,568 @@
+  In these cases, only a set of _trust anchors_ may be stored in VOA, while the _artifact verifiers_ are distributed with the signed artifacts.
+
+## File Hierarchy
+
+The _File Hierarchy for the Verification of OS Artifacts (VOA)_ is maintained as a directory structure on a system.
+It contains cryptographic public-key material for one or more technologies.
+
+The VOA hierarchy organizes _signature verifiers_ by [os], [purpose], [context] and [technology], and stores them in a directory structure of the form `$os/$purpose/$context/$technology/`.
+
+Vendors provide a set of _signature verifier_ files for their OS, in the VOA hierarchy format.
+
+Users of a VOA hierarchy (such as package installation software) can pick the relevant set of _signature verifiers_ for the verification of a specific artifact based on their location in the hierarchy.
+
+### Load paths
+
+VOA defines a list of _load paths_ with descending priority for system mode and user mode.
+
+_Signature verifiers_ are retrieved from hierarchies in these directories to validate artifacts based on specific [load logic].
+This is referred to as _verifier lookup_.
+
+The existence of multiple load paths allows system administrators and users to define custom sets of _signature verifiers_ for their system and applications.
+
+#### System mode
+
+The priority of load paths in system mode follows the definition of ["Storage Directories and Overrides" in the Configuration Files Sepcification], which is also reflected in the following list of paths:
+
+- `/etc/voa/`
+- `/run/voa/`
+- `/usr/local/share/voa/`
+- `/usr/share/voa/`
+
+#### User mode
+
+The priority of load paths in user mode mirrors that of the system mode.
+Refer to the [XDG Base Directory Specification] for details on default values.
+
+- `$XDG_CONFIG_HOME/voa/`
+- the `./voa/` directory in each directory defined in `$XDG_CONFIG_DIRS`
+- `$XDG_RUNTIME_DIR/voa/`
+- `$XDG_DATA_HOME/voa/`
+- the `./voa/` directory in each directory defined in `$XDG_DATA_DIRS`
+
+### Symlinking
+
+Symlinks can be used in the VOA hierarchy to point to files or directories below one of the [load paths].
+Symlinking to files or directories external to the load paths is not supported.
+
+### Masking
+
+Individual _artifact verifiers_ and _trust anchors_ may be masked using a symlink to `/dev/null`, independent of [technology].
+This constitutes the only allowed exception to the rule of not symlinking to files or directories external to the load paths.
+
+These symlinks are only expected to be used in writable and persistent load paths (that is, only below `/etc/voa/` when in system mode, or below`$XDG_DATA_HOME/voa/` when in user mode).
+Masked verifiers are not considered when retrieving from any of the VOA [load paths].
+
+Masking _artifact verifiers_ and _trust anchors_ may be desirable in situations where a verifier is considered compromised, but a revocation for it can not be provided to users in time.
```

Comment:
I wonder if we should clarify behavior in read-only file systems more, would these constraints be problematic? In conjunction with https://github.com/uapi-group/specifications/pull/134/files#diff-c79d1da1ef2fcfffc28ac7308505535eac0942d086d54a990553374ac81fed00R115 there would be ways to work with such filesystems. But symlinking to `/dev/null` might be problematic.

Reply:
Ah yes, similar to how `systemctl mask` can work on `--system` or `--runtime`, we can both use e.g. `/etc/voa/` or `/run/voa/`.

We have clarified the masking section and added a section for implementers about this.